service/history/shard/controller_impl.go

```diff
@@ -360,8 +361,11 @@ func (c *ControllerImpl) acquireShards() {
 	}
```

Comment:
btw, I just noticed that that `ctx` in `c.GetShardByID(ctx, shardID)` is ignored, so the `1*time.Second` timeout there doesn't do anything. we should either remove it or fix it so it works